src/main/java/net/dv8tion/jda/internal/handle/GuildSetupController.java

```diff
@@ -399,6 +383,23 @@ public void onUnavailable(long id)
         log.debug("Guild with id {} is now marked unavailable. Total: {}", id, unavailableGuilds.size());
     }
 
+    public void onTimeout()
```

Comment:
Should this method clear the `timeoutHandle` so that it can be set again via `startTimeout` if new guilds attempt to be loaded?

Reply:
Oh, I guess we are relying on the implicit clearing of `timeoutHandle` in `checkReady` due to us setting `incompleteCount` to `0` just before calling it.

Reply:
Yeah checkReady will clear it